src/main/java/co/cask/hydrator/salesforce/SObjectDescriptor.java

```diff
@@ -47,18 +47,26 @@
    * @return sObject descriptor
    * @throws ConnectionException in case of errors when establishing connection to Salesforce
    */
-  public static SObjectDescriptor fromName(String name, AuthenticatorCredentials credentials)
+  public static SObjectDescriptor fromName(String name,
+                                           AuthenticatorCredentials credentials, Set<FieldType> typesToSkip)
     throws ConnectionException {
     PartnerConnection partnerConnection = SalesforceConnectionUtil.getPartnerConnection(credentials);
     SObjectsDescribeResult describeResult = new SObjectsDescribeResult(
       partnerConnection, Collections.singletonList(name));
     List<FieldDescriptor> fields = describeResult.getFields().stream()
+      .filter(field -> !typesToSkip.contains(field.getType()))
```

Comment:
Please update java doc explaining why we might need to skip certain types.

Reply:
There is a comment when this is used. It is specific to streaming API at SalesforceStreamingSourceConfig#getSObjectQuery 
I don't think this specific streaming information should be added to general class.

However I changed javadoc to describe typesToSkip @param